### DISCUSSION TITLE
How to attach multiple routes to same pages component

### DISCUSSION THREAD

**Discussion by delucca** [vercel/next.js#18565] [Help]
Hi!

I'm developing an app that uses internationalization and I need to serve the same page in multiple routes.

For example, the route **getbud.co/resultados-chave** and **getbud.co/key-results** must both serve the same component located at `pages/myKeyResults.tsx`.

That is important for us since SEO is sensitive for the page URL (so, we cannot serve a Brazillian page called `key-results`, that would hurt our SEO score).

There is a proper way to do this? I've seen some options using `next-routes` or `next-routes-with-locale`, but both libraries seem to be extremely outdated (more than a year since the last release) and not an elegant solution.

I'm using `Koa` as my custom server, with the following `server.ts` file:

```typescript
import Koa from 'koa'
import Router from '@koa/router'
import next from 'next'
import { cyan, gray } from 'chalk'

import config from './config'

const launch = (): void => {
  const app = next({
    dev: config.dev,
  })

  app.prepare().then(() => {
    const server = new Koa()
    const router = new Router()
    const log = console.log
    const handle = app.getRequestHandler()

    server.use(async (ctx, next) => {
      ctx.res.statusCode = 200
      await next()
    })

    router.all('(.*)', async (ctx) => {
      await handle(ctx.req, ctx.res)
      ctx.respond = false
    })

    server.use(router.routes())

    server.listen(config.port, () =>
      log(`${cyan('➤')} ${gray('Web server running on:')} ${config.host}:${config.port}`),
    )
  })
}

export default {
  launch,
}
```

I had an idea to use the `router.get` to handle this logic for me, creating groups of routes and using the `app.render` to render a different component, but I'm using the new v10 i18n solution of Next ([ref](https://nextjs.org/docs/advanced-features/i18n-routing), which is AMAZING, btw) and it seems that the `getRequestHandler` does something to attach the `locale` to the request. If I use the `render` method from `app` the locale isn't attached to my request.

Anyone can give me a hand?

**Comment by delucca** [comment]
Taking a look at new examples, I found this: https://github.com/vercel/next.js/blob/canary/examples/custom-routes-proxying/next.config.js

It seems that I can proxy requests! Amazing :)

I did something like this:

```js
const serverRuntimeConfig = {}

const publicRuntimeConfig = {
  environment: process.env.NODE_ENV,
}

const i18n = {
  locales:
    publicRuntimeConfig.environment === 'production'
      ? ['pt-BR', 'en-US']
      : [process.env.LOCALE_OVERRIDE || 'pt-BR'],
  defaultLocale: 'pt-BR',

  domains: [
    {
      domain: 'getbud.co',
      defaultLocale: 'pt-BR',
    },
    {
      domain: 'en.getbud.co',
      defaultLocale: 'en-US',
    },
  ],
}

const i18nRouteGroup = [
  {
    destination: '/myKeyResults',
    variants: [
      '/resultados-chave',
      '/key-results',
    ],
  },
]

const buildProxiedRoutes = (...routeGroups) => routeGroups.map(buildSingleProxiedRouteGroup).flat(4)

const buildSingleProxiedRouteGroup = (routeGroup) => routeGroup.map(buildSingleRoute)

const buildSingleRoute = ({ destination, variants }) => variants.map((variant) => buildRouteVariant(variant, destination))

const buildRouteVariant = (variant, destination) => ({
  destination,
  source: variant,
})

module.exports = {
  serverRuntimeConfig,
  publicRuntimeConfig,
  i18n,

  async rewrites() { return buildProxiedRoutes(i18nRouteGroup) },
}
```

I'm going to do that, but still, I would like to hear from more experienced NextJS users if this is the proper solution for my problem.

**Comment by landsman** [comment]
We had to do this middleware: https://github.com/trisbee/next-i18n-routes-middleware for our sites and still looking forward for official way how to achive simillar results without custom SSR.
